J9 Java [Linux ARM] 80 Compile error: cannot declare member function 'static TR::Register* J9::ARM::TreeEvaluator::evaluateNULLCHKWithPossibleResolve(TR::Node*, bool, TR::CodeGenerator*)' to have static linkage [-fpermissive]

Failure link
------------


Optional info
-------------

Failure output (captured from console output)
---------------------------------------------

```
arm-bcm2708hardfp-linux-gnueabi-g++ -std=c++0x -fno-rtti -fno-threadsafe-statics -Wno-deprecated -Wno-enum-compare -Wno-invalid-offsetof -Wno-write-strings -pthread -fomit-frame-pointer -fasynchronous-unwind-tables -Wreturn-type -fno-strict-aliasing -fstack-protector -fPIC -mfloat-abi=hard -mfpu=vfp -march=armv6 -marm -O3   -DBITVECTOR_BIT_NUMBERING_MSB -DJ9_PROJECT_SPECIFIC -DTR_HOST_ARM -DTR_HOST_32BIT -DLINUX -DTR_TARGET_ARM -DTR_TARGET_32BIT -DSUPPORTS_THREAD_LOCAL -D_LONG_LONG -DARMGNU -DARMGNUEABI -DFIXUP_UNALIGNED -DHARDHAT -I'/tmp/bld_433827/bld_linux_arm/compiler/../omr/include_core' -I'/tmp/bld_433827/bld_linux_arm/compiler/../omr/thread' -I'/tmp/bld_433827/bld_linux_arm/compiler/../omr/gc/include' -I'/tmp/bld_433827/bld_linux_arm/compiler/../compiler/arm/' -I'/tmp/bld_433827/bld_linux_arm/compiler/../compiler/arm' -I'/tmp/bld_433827/bld_linux_arm/compiler/../compiler' -I'/tmp/bld_433827/bld_linux_arm/compiler/../omr/compiler/arm/' -I'/tmp/bld_433827/bld_linux_arm/compiler/../omr/compiler/arm' -I'/tmp/bld_433827/bld_linux_arm/compiler/../omr/compiler' -I'/tmp/bld_433827/bld_linux_arm/compiler/../omr' -I'/tmp/bld_433827/bld_linux_arm/compiler/..' -I'/tmp/bld_433827/bld_linux_arm/compiler/../codert_vm' -I'/tmp/bld_433827/bld_linux_arm/compiler/../gc_include' -I'/tmp/bld_433827/bld_linux_arm/compiler/../include' -I'/tmp/bld_433827/bld_linux_arm/compiler/../gc_glue_java' -I'/tmp/bld_433827/bld_linux_arm/compiler/../jit_vm' -I'/tmp/bld_433827/bld_linux_arm/compiler/../nls' -I'/tmp/bld_433827/bld_linux_arm/compiler/../oti' -I'/tmp/bld_433827/bld_linux_arm/compiler/../util' -MMD -MF /tmp/bld_433827/bld_linux_arm/compiler/../objs/compiler/control/J9Recompilation.o.depend.mk -MT /tmp/bld_433827/bld_linux_arm/compiler/../objs/compiler/control/J9Recompilation.o -MP -o /tmp/bld_433827/bld_linux_arm/compiler/../objs/compiler/control/J9Recompilation.o -c /tmp/bld_433827/bld_linux_arm/compiler/../compiler/control/J9Recompilation.cpp
In file included from /tmp/bld_433827/bld_linux_arm/compiler/../oti/j9.h:45:0,
                 from /tmp/bld_433827/bld_linux_arm/compiler/../compiler/env/RawAllocator.hpp:43,
                 from /tmp/bld_433827/bld_linux_arm/compiler/../compiler/env/PersistentAllocator.hpp:33,
                 from /tmp/bld_433827/bld_linux_arm/compiler/../omr/compiler/env/TRMemory.hpp:67,
                 from /tmp/bld_433827/bld_linux_arm/compiler/../omr/compiler/infra/List.hpp:27,
                 from /tmp/bld_433827/bld_linux_arm/compiler/../omr/compiler/env/FrontEnd.hpp:38,
                 from /tmp/bld_433827/bld_linux_arm/compiler/../omr/compiler/codegen/OMRCodeGenerator.hpp:39,
                 from /tmp/bld_433827/bld_linux_arm/compiler/../omr/compiler/arm/codegen/OMRCodeGenerator.hpp:36,
                 from /tmp/bld_433827/bld_linux_arm/compiler/../compiler/codegen/J9CodeGenerator.hpp:36,
                 from /tmp/bld_433827/bld_linux_arm/compiler/../compiler/arm/codegen/J9CodeGenerator.hpp:40,
                 from /tmp/bld_433827/bld_linux_arm/compiler/../compiler/codegen/CodeGenerator.hpp:26,
                 from /tmp/bld_433827/bld_linux_arm/compiler/../compiler/control/J9Recompilation.cpp:24:
/tmp/bld_433827/bld_linux_arm/compiler/../include/jni.h:461:10: note: the mangling of 'va_list' has changed in GCC 4.4
/tmp/bld_433827/bld_linux_arm/compiler/../compiler/arm/codegen/J9TreeEvaluator.cpp:69:116: error: cannot declare member function 'static TR::Register* J9::ARM::TreeEvaluator::evaluateNULLCHKWithPossibleResolve(TR::Node*, bool, TR::CodeGenerator*)' to have static linkage [-fpermissive]
/tmp/bld_433827/bld_linux_arm/compiler/../compiler/arm/codegen/J9UnresolvedDataSnippet.cpp: In member function 'virtual uint8_t* J9::ARM::UnresolvedDataSnippet::emitSnippetBody()':
/tmp/bld_433827/bld_linux_arm/compiler/../compiler/arm/codegen/J9UnresolvedDataSnippet.cpp:107:70: warning: cast to pointer from integer of different size [-Wint-to-pointer-cast]
make[3]: *** [/tmp/bld_433827/bld_linux_arm/compiler/../objs/compiler/arm/codegen/J9TreeEvaluator.o] Error 1
make[3]: *** Waiting for unfinished jobs....
In file included from /tmp/bld_433827/bld_linux_arm/compiler/../compiler/control/J9Recompilation.cpp:25:0:
/tmp/bld_433827/bld_linux_arm/compiler/../compiler/control/CompilationRuntime.hpp: In static member function 'static uint32_t TR::CompilationInfo::getJ9MethodJITExtra(J9Method*)':
/tmp/bld_433827/bld_linux_arm/compiler/../compiler/control/CompilationRuntime.hpp:524:54: warning: right shift count >= width of type [enabled by default]
make[3]: Leaving directory `/tmp/bld_433827/bld_linux_arm/compiler'
make[2]: *** [default] Error 2
make[2]: Leaving directory `/tmp/bld_433827/bld_linux_arm/compiler'
make[1]: *** [j9jitlauncher] Error 2
make[1]: Leaving directory `/tmp/bld_433827/bld_linux_arm'
make: *** [phase_jit] Error 2
```

--- a/runtime/compiler/arm/codegen/J9TreeEvaluator.cpp
+++ b/runtime/compiler/arm/codegen/J9TreeEvaluator.cpp
@@ -65,7 +65,7 @@ J9::ARM::TreeEvaluator::resolveAndNULLCHKEvaluator(TR::Node *node, TR::CodeGener
    return TR::TreeEvaluator::evaluateNULLCHKWithPossibleResolve(node, true, cg);
    }
 
-static TR::Register *
+TR::Register *
 J9::ARM::TreeEvaluator::evaluateNULLCHKWithPossibleResolve(TR::Node *node, bool needsResolve, TR::CodeGenerator *cg)
    {
    // NOTE:

